topics/basics/testing_plugins/integration_tests/integration_tests_intro.md

```diff
@@ -0,0 +1,286 @@
+
+### 2. Plugin installation
+
+```kotlin
+.apply {
+  val pathToPlugin = System.getProperty("path.to.build.plugin")
+  PluginConfigurator(this).installPluginFromPath(Path(pathToPlugin))
+}
+```
+
+This step configures plugin installation using the plugin path we defined in our Gradle configuration.
+
+> `PluginConfigurator` can install plugins from local paths or Marketplace.
+>
+{style="note"}
+
+### 3. IDE life cycle management
+
+```kotlin
+.runIdeWithDriver().useDriverAndCloseIde {
+}
+```
+
+These two methods:
+
+* Start the IDE instance (`runIdeWithDriver`).
+* Shut down the IDE (`useDriverAndCloseIde`).
+
+The empty lambda is used for IDE interactions (`useDriverAndCloseIde`).
+
+> When the test is run for the first time, it may take longer than expected since it needs to download the IDE. Subsequent runs will be faster, using the cached IDE version.
+>
+{style="note"}
+
+## Understanding the Test Architecture
+
+The test can now be run and it should pass.
+
+Integration tests operate across two separate processes:
+
+* Test process:
+    * Executes test code, sending commands to the IDE.
+    * Manages the IDE life cycle.
+    * Controls test flow and assertions.
+* IDE process:
+    * Listens and executes commands from the test process.
+
+This dual-process architecture explains several key aspects of integration testing:
+
+* Why debugging requires special considerations.
+* The need for a communication protocol between test and IDE processes.
+* Why a built plugin distribution is required.
+* The origin of certain test-specific exceptions.
+
+## Opening Projects in Tests
+
+While starting an IDE with an empty project is useful, sometimes we need actual projects to verify real-world scenarios. Let's modify our test to open a project.
+
+The framework supports several ways to specify test projects:
+
+* **GitHub projects**: `GitHubProject.fromGithub( branchName = "master", "JetBrains/ij-perf-report-aggregator" )`
+* **Remote archives**: `RemoteArchiveProjectInfo("https://github.com/JetBrains/intellij-community/archive/master.zip")`
+* **Local projects**: `LocalProjectInfo(Path("src/test/resources/test-projects/simple-project"))`
+
+Here's our test modified to open a GitHub project:
+
+```kotlin
+@Test
+fun simpleTest() {
+  Starter.newContext(
+    "testExample",
+    TestCase(
+      IdeProductProvider.IC,
+      GitHubProject.fromGithub(branchName = "master", repoRelativeUrl = "JetBrains/ij-perf-report-aggregator")
+    ).withVersion("2024.2")
+  ).apply {
+    val pathToPlugin = System.getProperty("path.to.build.plugin")
+    PluginConfigurator(this).installPluginFromPath(Path(pathToPlugin))
+  }.runIdeWithDriver().useDriverAndCloseIde {
+    waitForIndicators(5.minutes)
+  }
+}
+```
+
+While simple, this test verifies a critical aspect: our plugin doesn't interfere with IDE startup.
+
+> We added a call `waitForIndicators` into the lambda to make sure that we wait till all indicators are gone before exiting the IDE.
```

Comment:
```suggestion
> Notice a `waitForIndicators` call in the lambda.
> It ensures that the test will wait till all indicators are gone before exiting the IDE.
```

Not sure about "the test will wait". But likely it is not "we" who is waiting here.